### ISSUE TITLE
salt/utils/cloud.py calls a non-existent method in pywinrm, resulting in failures for winrm bootstrapped host.

### ISSUE THREAD

**Issue by bensherman** [saltstack/salt#34008]
### Description of Issue/Question
salt/utils/cloud.py calls a non-existent method in pywinrm, resulting in failures for winrm bootstrapped host.

### Setup
Try and create a windows 2012 r2 host in ec2 with winrm enabled in the salt cloud profile.

### More info
In salt/utils/cloud.py (develop and 2016.3 branches)
Line 835 says:
s.protocol.set_timeout(15)

winrm.Session.protocol has no set_timeout method.  There are two properties that are set to sane defaults, perhaps this can just go away.

    DEFAULT_READ_TIMEOUT_SEC = 30
    DEFAULT_OPERATION_TIMEOUT_SEC = 20
    read_timeout_sec=DEFAULT_READ_TIMEOUT_SEC, 
    operation_timeout_sec=DEFAULT_OPERATION_TIMEOUT_SEC, 

### Versions Report
```
Salt Version:
           Salt: 2016.3.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.18.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

**Comment by Ch3LL** [comment]
@bensherman thanks for the PR! We will get this closed once the PR is merged. 

Also ping @alxbse  looks like you were the one that added this so wanted you to be aware of this.

**Comment by bensherman** [comment]
Looks like that method was removed recently, added some logic to set it if it exists.  See commit in pywinrm at https://github.com/diyan/pywinrm/commit/1213eaffd72e0276b52e46461c0d4a4c4fd8878a

**Comment by rallytime** [comment]
Fixed via #34026
